crates/store/src/db/models/queries/transactions.rs

```diff
@@ -126,14 +173,191 @@ pub struct TransactionSummaryRowInsert {
+            output_notes: output_notes_binary,
+            size_in_bytes,
+        }
+    }
+}
+
+/// Select complete transaction records for the given accounts and block range.
+///
+/// # Parameters
+/// * `account_ids`: List of account IDs to filter by
+///     - Limit: 0 <= size <= 1000
+/// * `block_range`: Range of blocks to include inclusive
+///
+/// # Returns
+/// A tuple of (`last_block_included`, `transaction_records`) where:
+/// - `last_block_included`: The highest block number included in the response
+/// - `transaction_records`: Vector of transaction records, limited by payload size
+///
+/// # Note
+/// This function returns complete transaction record information including state commitments
+/// and note IDs, allowing for direct conversion to proto `TransactionRecord` without loading
+/// full block data. We use a chunked loading strategy to prevent memory exhaustion attacks and
+/// ensure predictable resource usage.
+///
+/// # Raw SQL
+/// ```sql
+/// SELECT
+///     account_id,
+///     block_num,
+///     transaction_id,
+///     initial_state_commitment,
+///     final_state_commitment,
+///     input_notes,
+///     output_notes,
+///     size_in_bytes
+/// FROM transactions
+/// WHERE block_num > ?1 AND block_num <= ?2 AND account_id IN rarray(?3)
+/// ORDER BY block_num ASC
+/// LIMIT ?4 OFFSET ?5
+/// ```
+/// Note: The query is executed in chunks of 1000 transactions to prevent loading
+/// excessive data and to stop as soon as the accumulated size approaches the 5MB limit.
+pub fn select_transactions_records(
+    conn: &mut SqliteConnection,
+    account_ids: &[AccountId],
+    block_range: RangeInclusive<BlockNumber>,
+) -> Result<(BlockNumber, Vec<crate::db::TransactionRecord>), DatabaseError> {
+    const MAX_PAYLOAD_BYTES: usize = 5 * 1024 * 1024; // 5 MB
+    const CHUNK_SIZE: i64 = 1000; // Read 1000 transactions at a time
+
+    QueryParamAccountIdLimit::check(account_ids.len())?;
+
+    if block_range.is_empty() {
+        return Err(DatabaseError::InvalidBlockRange {
+            from: *block_range.start(),
+            to: *block_range.end(),
+        });
+    }
+
+    let desired_account_ids = serialize_vec(account_ids);
+
+    // Read transactions in chunks to prevent loading excessive data and to stop
+    // as soon as we approach the size limit
+    let mut all_transactions = Vec::new();
+    let mut current_size = 0i64;
+    let mut offset = 0i64;
+
+    loop {
+        let chunk =
+            SelectDsl::select(schema::transactions::table, TransactionRecordRaw::as_select())
+                .filter(schema::transactions::block_num.gt(block_range.start().to_raw_sql()))
+                .filter(schema::transactions::block_num.le(block_range.end().to_raw_sql()))
+                .filter(schema::transactions::account_id.eq_any(&desired_account_ids))
+                .order((schema::transactions::block_num.asc(),))
+                .offset(offset)
+                .limit(CHUNK_SIZE)
```

Comment:
Since we are using limit + offset here, there a few things to consider:

1. We need to make sure that all queries happen in the same transaction (I think that's already the case).
2. I think to ensure stable order, we also need to sort by transaction ID (i.e., just sorting by block number may be insufficient).
3. From what I know, `offset` may not be all that efficient in SQLite (cc @Mirko-von-Leipzig as he may know more on this) - and I wonder if there is a better way to structure the query that will rely on a more explicit condition.

Reply:
1. The call is wrapped in a DB transaction:
```rust
        self.transact("full transactions records", move |conn| {
            queries::select_transactions_records(conn, &account_ids, block_range)
        })
        .await
```
2. I added the order by transaction ID too.
3. For this point, maybe we can go with some cursors using the block numbers & transaction ID instead of the `OFFSET`, but I'm not entirely sure what's best on SQLite.